src/vcModals.cpp

```diff
@@ -746,6 +749,131 @@ void vcModals_DrawUnsupportedFiles(vcState *pProgramState)
   }
 }
 
+bool vcModals_TakeScreenshot(vcState *pProgramState, char *pFilename, uint32_t bufLen, int *pSuffix)
+{
+  if (pFilename == nullptr || pProgramState == nullptr)
+    return false;
+
+  if (pProgramState->screenshot.resize)
+  {
+    pProgramState->sceneResolution.x = resses[pProgramState->screenshot.res].x;
+    pProgramState->sceneResolution.y = resses[pProgramState->screenshot.res].y;
+
+    pProgramState->screenshot.resize = false;
+    return true;
+  }
+  else if (pProgramState->image.pImage == nullptr)
+  {
+    return true;
+  }
+
+  if (pFilename == "")
+    udStrcpy(pFilename, bufLen, vcString::Get("screenshotDefaultName"));
+
+  pProgramState->screenshot.pPixels = udAllocType(uint8_t, pProgramState->sceneResolution.x * pProgramState->sceneResolution.y * 4, udAF_Zero);
+
+  vcTexture_BeginReadPixels(pProgramState->image.pImage, 0, 0, pProgramState->sceneResolution.x, pProgramState->sceneResolution.y, pProgramState->screenshot.pPixels, pProgramState->pRenderContext->pFramebuffer[1]);
```

Comment:
It would be useful to `vcTexture_SaveToFile` to have this functionality rather than doing it in a modal.

Reply:
This will be a little complex as we need the valid attached framebuffer to do so in this way, for earlier versions of opengl